crates/sui-types/src/committee.rs

```diff
@@ -357,6 +359,7 @@ pub trait CommitteeTrait<K: Ord> {
 pub struct NetworkMetadata {
     pub network_address: Multiaddr,
     pub narwhal_primary_address: Multiaddr,
+    pub network_public_key: NetworkPublicKey,
```

Comment:
Why does this struct support `Serialize, Deserialize`? And why do we need this additional field? We are making an incompatible change and I want to be sure it is safe.

Reply:
removed `Serialize` and `Deserialize`

Reply:
we need the additional field in order to generate the TLS client config after enabling the commented-out code in `make_network_authority_clients_with_network_config`